### PR TITLE
fix(ci): Fix `linux_train.py` "logits" failure

### PR DESCRIPTION
**Pull request by courtneypacheco**
**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

In certain situations, `linux_train.py` fails with: "UserWarning: `return_dict_in_generate` is NOT set to `True`, but `output_logits` is. When `return_dict_in_generate` is not `True`, `output_logits` is ignored."

To fix this failure, I have added `return_dict_in_generate=True` as an input to `model.generate()`.

### CODE REVIEW

**Comment by RobotSail** [comment]
@courtneypacheco We have a PR to fix this in #3264

**Comment by booxter** [comment]
Please link to https://github.com/instructlab/instructlab/issues/3289 if you intend to continue on this PR to fix the issue in linux-train flow.

**Comment by booxter** [comment]
CI issue should be addressed with https://github.com/instructlab/instructlab/pull/3307 I believe. If we still have issues with the job, please reopen and refresh the patch. For now, I'm closing it.
